crates/stackable-versioned-macros/fixtures/README.md

```diff
@@ -0,0 +1,54 @@
+# Snapshot Testing
+
+This folder contains fixtures for snapshot testing the `#[versioned()]` macro. Snapshot testing is
+done using the [insta] crate. It provides a [CLI tool][insta-cli] called `cargo-insta` and a
+[VS Code extension][insta-ext].
+
+Test inputs and snapshots of the expected output are located in the `fixtures` folder. There are two
+inputs to the `#[versioned()]` macro because it is an attribute macro:
+
+> The first TokenStream is the delimited token tree following the attribute’s name, not including
+> the outer delimiters. If the attribute is written as a bare attribute name, the attribute
+> TokenStream is empty. The second TokenStream is the rest of the item including other attributes on
+> the item.
+>
+> _(Taken from the [Rust reference][rust-ref])_
+
+Because of that, a special delimiter is used in the input files which separates the two inputs while
+still enabling developers to write valid Rust code. The delimiter is `// ---\n`. Most of the inner
+workings are located in [this file](../src/test_utils.rs).
+
+```rust
+#[versioned(
+    version(name = "v1alpha1"),
+    version(name = "v1beta1"),
+    version(name = "v1")
+)]
+// --- <- See here!
+pub(crate) struct Foo {
+    #[versioned(
+        changed(since = "v1beta1", from_name = "jjj", from_type = "u8"),
+        changed(since = "v1", from_type = "u16"),
+    )]
+    bar: usize,
+    baz: bool,
+}
+```
+
+## Recommended Workflow
+
+First, add new input files (which automatically get picked up by `insta`) to the `fixtures/inputs`
+folder. Make sure the delimiter is placed correctly between the attribute and the container
+definition. Next, generate the snapshot files (initially not accepted) by running
+
+```shell
+cargo insta test -p stackable-versioned-macros
+```
+
+This command will place the new snapshot files in the `fixtures/snapshots` folder. To review them,
```

Comment:
Could be worth mentioning that they are suffixed with `.new`, and should not be committed unless internationally (eg: for collaboration). They should not appear on the `main` branch.

Reply:
Added in cf1ec21